### PR TITLE
Data Stores: Refactor `ProductsList` store to `createReduxStore()`

### PR DESCRIPTION
**Pull request by tyxla**
## Proposed Changes

This PR migrates the `ProductsList` store to use `createReduxStore()` and `register()` instead of `registerStore()`.

Part of #74399. A follow-up to #73890.

## Testing Instructions

* Domain selection - the domains and prices still appear properly.
* Design selection - the modal that appears when selecting a paid theme and the button to unlock it.
* Sensei domain selection - starting on `/setup/sensei/` you can properly pick a domain and continue with the flow.
* Sensei plan selection - plan step looks well and you can properly pick a plan.
* Woo confirm - starting with a free plan, the step after selecting that you want to sell on your site works well.
* Verify all checks are green.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts
Comment (gikaragia) [inline client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts:1]: Tested Sensei flow, it works well 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
 import { useSelect } from '@wordpress/data';
 import { useSupportedPlans } from 'calypso/../packages/plans-grid/src/hooks';
-import { PLANS_STORE, PRODUCTS_LIST_STORE } from 'calypso/landing/stepper/stores';
-import type {
-	PlanBillingPeriod,
-	PlansSelect,
-	ProductsListSelect,
-} from 'calypso/../packages/data-stores';
+import { PLANS_STORE } from 'calypso/landing/stepper/stores';
+import type { PlanBillingPeriod, PlansSelect } from 'calypso/../packages/data-stores';
```
Comment (noahtallen) [inline client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts]: While we're here, could we change this to import from `@automattic/data-stores`?

**Review by noahtallen — APPROVED** [review]
Tests well for me!

File: client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts (hunk 1 above)
Comment (tyxla) [reply client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/sensei-plan-products.ts]: Good call - done in 765618900a235619bc9f7969ac1859c334fd50f3.

**Pull request merged**